### ISSUE TITLE
straight-vc-git--ensure-head error

### ISSUE THREAD

**Issue by bolasblack** [raxod502/straight.el#306]
This is my recipe:

```elisp
(use-package beancount-mode
  :straight (beancount :host github
                       :repo "beancount/beancount"
                       :files ("editors/emacs/*.el")))
```

When I executed `straight-freeze-versions`, straight warn me:

![image](https://user-images.githubusercontent.com/382011/45253363-80141200-b398-11e8-8954-a398aee975fd.png)

Then I checked the repo, it not ahead

Then I read the code

https://github.com/raxod502/straight.el/blob/e65a2e22764552b692fd12d82ce7d1605765eed8/straight.el#L1628-L1630

https://github.com/raxod502/straight.el/blob/e65a2e22764552b692fd12d82ce7d1605765eed8/straight.el#L819-L822

And tried `git merge-base --is-ancestor master HEAD` and get a warning: `warning: refname 'master' is ambiguous.`, seems this warning occurred `straight--check-call` think the repo is ahead

**Comment by bolasblack** [comment]
Seems there is a tag named master

![image](https://user-images.githubusercontent.com/382011/45254218-53b3c200-b3a7-11e8-997e-a2039356865c.png)

**Comment by bolasblack** [comment]
Sorry about it, I noticed `straight--check-call` only check the exit code, and `git merge-base --is-ancestor master HEAD`'s exit code is 0. But `git merge-base --is-ancestor HEAD master` truly returned 1.

**Comment by raxod502** [comment]
Yes, this code is fragile and should be improved. Thanks for the report. Contributions are welcome, since I expect to be very busy for the next few months.

**Comment by progfolio** [comment]
Fixed by #711
